server/src/main/java/org/opensearch/cluster/coordination/PersistedStateStats.java

```diff
@@ -0,0 +1,126 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.cluster.coordination;
+
+import org.opensearch.core.common.io.stream.StreamInput;
+import org.opensearch.core.common.io.stream.StreamOutput;
+import org.opensearch.core.common.io.stream.Writeable;
+import org.opensearch.core.xcontent.ToXContentObject;
+import org.opensearch.core.xcontent.XContentBuilder;
+
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.concurrent.atomic.AtomicLong;
+
+/**
+ * Persisted cluster state related stats.
+ *
+ * @opensearch.internal
+ */
+public class PersistedStateStats implements Writeable, ToXContentObject {
+    private String statsName;
+    private AtomicLong totalTimeInMillis = new AtomicLong(0);
+    private AtomicLong failedCount = new AtomicLong(0);
+    private AtomicLong successCount = new AtomicLong(0);
+    private Map<String, AtomicLong> extendedFields = new HashMap<>(); // keeping minimal extensibility
+
+    public PersistedStateStats(String statsName) {
+        this.statsName = statsName;
+    }
+
+    @Override
+    public void writeTo(StreamOutput out) throws IOException {
+        out.writeVLong(successCount.get());
+        out.writeVLong(failedCount.get());
+        out.writeVLong(totalTimeInMillis.get());
+        if (extendedFields.size() > 0) {
+            out.writeBoolean(true);
+            out.writeVInt(extendedFields.size());
+            for (Map.Entry<String, AtomicLong> extendedField : extendedFields.entrySet()) {
+                out.writeString(extendedField.getKey());
+                out.writeVLong(extendedField.getValue().get());
+            }
+        } else {
+            out.writeBoolean(false);
+        }
+    }
+
+    public PersistedStateStats(StreamInput in) throws IOException {
+        this.successCount = new AtomicLong(in.readVLong());
+        this.failedCount = new AtomicLong(in.readVLong());
+        this.totalTimeInMillis = new AtomicLong(in.readVLong());
+        if (in.readBoolean()) {
+            int extendedFieldsSize = in.readVInt();
+            this.extendedFields = new HashMap<>();
+            for (int fieldNumber = 0; fieldNumber < extendedFieldsSize; fieldNumber++) {
+                extendedFields.put(in.readString(), new AtomicLong(in.readVLong()));
+            }
+        }
+    }
+
+    @Override
+    public XContentBuilder toXContent(XContentBuilder builder, Params params) throws IOException {
+        builder.startObject(statsName);
+        builder.field(Fields.UPDATE_COUNT, getSuccessCount());
```

Comment:
SUCCESS_COUNT

Reply:
Changing field name also to **success_count** and constant name to SUCCESS_COUNT